src/mappings/eventHandlers.ts

```diff
@@ -74,12 +77,19 @@ async function findOrCreateBlock(substrateBlock: SubstrateBlock): Promise<String
     return existingBaseBlock.id;
   }
 
+  // Calculate weight
+  let blockWeight = new BN(0);
+  events.forEach((event) => {
+    const { event: { method, section, data } } = event;
+    const weight = section === 'system' && ['ExtrinsicFailed', 'ExtrinsicSuccess'].includes(method) ? ((method === 'ExtrinsicSuccess' ? data[0] : data[1]) as DispatchInfo).weight : new BN(0);
```

Comment:
mind breaking this up into a separate weight calculation function where you pass in the event? This 1 liner is really confusing to read, so it's probably better if you swapped it out for some if statements and good variable naming

Reply:
Fixed!